src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMFile.java

```diff
@@ -108,8 +108,10 @@ private Object metadata() throws IOException {
                             info = TypeInfo.NON_DIRECTORY_CONFIRMED;
                             resolved = true;
                         } catch (IOException e) {
-                            if (e.getCause() instanceof IOException
-                                    && e.getCause().getCause() instanceof JsonMappingException) {
+                            // Upcoming version of github-api hoists JsonMappingException up one level
+                            // Support both the old and the new structure
+                            if (e.getCause() instanceof JsonMappingException
+                                || e.getCause().getCause() instanceof JsonMappingException ) {
```

Comment:
I think you need a null check here:

```suggestion
                                || e.getCause() != null && e.getCause().getCause() instanceof JsonMappingException ) {
```